src/rust/engine/process_execution/src/remote.rs

```diff
@@ -843,16 +842,6 @@ impl crate::CommandRunner for CommandRunner {
     )
     .await
   }
-
```

Comment:
So, one of the more important places where the MPP constraint actually used to be used... seemingly no longer exists. The speculating command runner used to skip running remotely if the remote platform didn't match the local platform.

I believe that that constraint application is still necessary... but I think that with https://docs.google.com/document/d/1hS0n6vp-hBxy4Xo-q9QA_elofzQtOubjfN4nSeIgp90/edit# , the relevant place to apply it might be in `node.rs` to explicitly choose between a remote and a local `CommandRunner` based on which matches the platform.

Having said that, there are a few gaps in here, and we're essentially starting over for cross-platform speculation. So fine for a followup.